tests/FSharp.Compiler.ComponentTests/Miscellaneous/FsharpSuiteMigrated.fs

```diff
@@ -0,0 +1,563 @@
+// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+module FSharp.Compiler.ComponentTests.Miscellaneous.FsharpSuiteMigrated
+
+open System
+open System.IO
+open Xunit
+open FSharp.Test
+open FSharp.Test.Compiler
+open FSharp.Test.ScriptHelpers 
+
+module ScriptRunner = 
+    open Internal.Utilities.Library
+    
+    let private createEngine(args,version) = 
+        let scriptingEnv = getSessionForEval args version
+        scriptingEnv.Eval """let exit (code:int) = if code=0 then () else failwith $"Script called function 'exit' with code={code}" """ |> ignore
```

Comment:
Typically, each .fsx test file ends with "exit 0" or "exit 1". 
This works nice in a standalone script execution, but is bad if executed in-process -> it crashed entire xunit test host without any clue why and which test caused it (so was also difficult to find in the first place).